plugin/src/main/java/org/opensearch/ml/cluster/MLSyncUpCron.java

```diff
@@ -168,37 +172,44 @@ public void run() {
             MLSyncUpInput syncUpInput = inputBuilder.build();
             MLSyncUpNodesRequest syncUpRequest = new MLSyncUpNodesRequest(allNodes, syncUpInput);
             // sync up running model/tasks on nodes
-            client
-                .execute(
-                    MLSyncUpAction.INSTANCE,
-                    syncUpRequest,
-                    ActionListener.wrap(re -> { log.debug("sync model routing job finished"); }, ex -> {
-                        log.error("Failed to sync model routing", ex);
-                    })
-                );
-            // Undeploy expired models
-            undeployExpiredModels(modelsToUndeploy, modelWorkerNodes);
+            client.execute(MLSyncUpAction.INSTANCE, syncUpRequest, ActionListener.wrap(re -> {
+                log.debug("sync model routing job finished");
+                if (!modelsToUndeploy.isEmpty()) {
+                    // Undeploy expired models
+                    undeployExpiredModels(modelsToUndeploy, modelWorkerNodes, deployingModels);
+                    return;
```

Comment:
Why return here ? If we have 10 models, `modelsToUndeploy` has 4 models. We should also call `refreshModelState` for the other 6, right?

Reply:
The refreshModelState is called in the listener of unDeployModels after the expired models are un-deployed. Check the undeployExpiredModels(), So the modelstate will be refreshed anyways.